server/sched.go

```diff
@@ -166,6 +167,17 @@ func (s *Scheduler) processPending(ctx context.Context) {
 						gpus = s.getGpuFn()
 					}
 
+					if pending.origNumCtx == -1 {
+						if len(gpus) == 1 && gpus[0].Library != "cpu" && gpus[0].TotalMemory <= 4096*1024*1024 {
```

Comment:
Awesome!

For smaller models like Gemma 3 1B, the full 32k context window _should_ fit on a 4GB GPU. I'm not sure how to handle that in this case (maybe also considering the model size, or checking if we'd go over the available VRAM with the 4k default?)

Reply:
Note: above consideration could be in a follow-up, but wanted to note it here

Reply:
that's a good point, that could be a really valuable special case. I think for this change we want to stick with things that are mostly "neutral", but we really should consider cases in which we're willing to be opportunistic based on the available VRAM and try to fast follow